forui/lib/src/widgets/button/button_spinner.dart

```diff
@@ -0,0 +1,153 @@
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+
+/// An animated spinner icon.
+///
+/// Typically used with an [FButton] as a prefixIcon. The spinner will rotate indefinitely.
+/// The spinner's color and size defaults to the parent button's [FButtonStyle].
+///
+/// See:
+/// * https://forui.dev/docs/form/button for working examples.
+/// * [FButton] for creating a button.
+/// * [FButtonCustomStyle] for customizing a button's appearance.
+class FButtonSpinner extends StatefulWidget {
+  /// The style. Defaults to the parent button's [FButtonSpinnerStyle].
+  ///
+  /// Typically used to change the size and animation duration of the spinner.
+  final FButtonSpinnerStyle? style;
+
+  /// Creates a button spinner.
+  const FButtonSpinner({
+    this.style,
+    super.key,
+  });
+
+  @override
+  State<FButtonSpinner> createState() => _FButtonSpinnerState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty('style', style));
+  }
+}
+
+class _FButtonSpinnerState extends State<FButtonSpinner> with SingleTickerProviderStateMixin {
+  late AnimationController _controller;
+  late Animation<double> _animation;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final FButtonData(:style) = FButtonData.of(context);
```

Comment:
nitpick: for single widgets, I think it's cleaner to assign it to a variable without using pattern matching.
```suggestion
    final style = FButtonData.of(context).style;
```